inst/dust/cows.cpp

```diff
@@ -355,4 +363,48 @@ class cows {
     }
     return dust2::zero_every_type<real_type>{{weekly, reset}};
   }
+
+  struct data_type {
+    std::vector<real_type> positive_tests;
+  };
+
+  static data_type build_data(cpp11::list r_data) {
+    auto data = static_cast<cpp11::list>(r_data);
+    // TODO: rich will change build_data to accept shared, from which
+    // we can read the number of regions.
+    const auto n_regions = 48;
+    std::vector<real_type> positive_tests(n_regions);
+    dust2::r::read_real_vector(r_data, n_regions, positive_tests.data(), "positive_tests", true);
+    return data_type{positive_tests};
+  }
+
+  static real_type compare_data(const real_type time,
```

Comment:
I'm not convinced the ll is being calculated correctly, but am struggling to know where to begin investigating. Is the "compare_data" function just pulled every time a data-point aligns with the time-step?

I think I could do with a dummy's guide to monty here!